### PR TITLE
Integrate scanpy for more efficient clustering and general functionality

### PR DESCRIPTION
**Pull request by adamgayoso**
Scanpy backend implemented

- use Scanpy Louvain clustering
- adds `tqdm` for progress bar
- validated on internal datasets
- default params run method as is in version 2.4.0
- updates umap plot by showing predicted singlets in grey, doublets in black

### CODE REVIEW

**Comment by JonathanShor** [comment]
Why did the verbosity branch get partially added in here?

**Comment by adamgayoso** [comment]
@JonathanShor I was trying to be pragmatic :(

**Comment by adamgayoso** [comment]
We could add back phonograph support as follows

```
import scanpy.external as sce
result = sce.tl.phenograph(adata.obsm['X_pca'], k = 30)
```

File: src/doubletdetection/doubletdetection.py [hunk 1]
```diff
@@ -300,23 +307,30 @@ def _one_fit(self):
...
 
-        print("Running PCA...")
-        # Get phenograph results
-        pca = PCA(n_components=self.n_components, random_state=self.random_state)
-        reduced_counts = pca.fit_transform(aug_counts)
-        print("Clustering augmented data set with Phenograph...\n")
-        fullcommunities, _, _ = phenograph.cluster(reduced_counts, **self.phenograph_parameters)
+        aug_counts = anndata.AnnData(aug_counts)
+        aug_counts.obs['n_counts'] = aug_lib_size
+        sc.pp.scale(aug_counts, max_value=15)
```
Comment (adamgayoso) [inline src/doubletdetection/doubletdetection.py]: We should make this line optional.

**Comment by adamgayoso** [comment]
> We could add back phonograph support as follows
> 
> ```
> import scanpy.external as sce
> result = sce.tl.phenograph(adata.obsm['X_pca'], k = 30)
> ```
> 
> In fact we should make it an option so that we can always use the original method. This would also involve making the standard scaling an option.

I added this, but not through the external scanpy module. I added back what we had before.

**Comment by adamgayoso** [comment]
@JonathanShor This is good to squash into dev. The only thing that is missing is that your PhenoGraph actually doesn't have a verbose or verbosity argument. So, if you choose to `use_phenograph` and verbose is `False` then Pheno will still print stuff out.

**Pull request merged**